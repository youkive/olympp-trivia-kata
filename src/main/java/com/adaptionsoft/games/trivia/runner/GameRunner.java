
package com.adaptionsoft.games.trivia.runner;

import com.adaptionsoft.games.uglytrivia.Game;

import java.io.PrintStream;
import java.util.Random;


public class GameRunner {

	private static boolean notAWinner;

	public static void main(String[] args) {
		PrintStream out = System.out;
		Random rand = new Random();
		run(out, rand.nextInt(5), rand.nextInt(9));
	}

	public static void run(PrintStream out, int randInt, int secondRandom) {
		Game aGame = new Game();

		aGame.add("Chet");
		aGame.add("Pat");
		aGame.add("Sue");

		do {

			aGame.roll(randInt + 1, out);

			if (secondRandom == 7) {
				notAWinner = aGame.wrongAnswer();
			} else {
				notAWinner = aGame.wasCorrectlyAnswered();
			}



		} while (notAWinner);
	}
}
