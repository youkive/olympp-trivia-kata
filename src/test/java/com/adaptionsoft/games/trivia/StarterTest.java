package com.adaptionsoft.games.trivia;

import com.adaptionsoft.games.trivia.runner.GameRunner;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

public class StarterTest {

    @Test
    void should_print_golden_source() {
        ByteArrayOutputStream testOutput = new ByteArrayOutputStream();
        PrintStream out = new PrintStream(testOutput);

        GameRunner.run(out,4, 8);

        Approvals.verify(testOutput.toString());
    }
}
