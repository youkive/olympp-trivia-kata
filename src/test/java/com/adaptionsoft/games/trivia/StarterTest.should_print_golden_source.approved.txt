Chet is the current player
They have rolled a 5
Chet's new location is 5
The category is Science
Pat is the current player
They have rolled a 5
Pat's new location is 5
The category is Science
Sue is the current player
They have rolled a 5
Sue's new location is 5
The category is Science
Chet is the current player
They have rolled a 5
Chet's new location is 10
The category is Sports
Pat is the current player
They have rolled a 5
Pat's new location is 10
The category is Sports
Sue is the current player
They have rolled a 5
Sue's new location is 10
The category is Sports
Chet is the current player
They have rolled a 5
Chet's new location is 3
The category is Rock
Pat is the current player
They have rolled a 5
Pat's new location is 3
The category is Rock
Sue is the current player
They have rolled a 5
Sue's new location is 3
The category is Rock
Chet is the current player
They have rolled a 5
Chet's new location is 8
The category is Pop
Pat is the current player
They have rolled a 5
Pat's new location is 8
The category is Pop
Sue is the current player
They have rolled a 5
Sue's new location is 8
The category is Pop
Chet is the current player
They have rolled a 5
Chet's new location is 1
The category is Science
Pat is the current player
They have rolled a 5
Pat's new location is 1
The category is Science
Sue is the current player
They have rolled a 5
Sue's new location is 1
The category is Science
Chet is the current player
They have rolled a 5
Chet's new location is 6
The category is Sports
